pkg/infinispan/grafana/dashboard_controller.go

```diff
@@ -0,0 +1,103 @@
+package grafana
+
+import (
+	"context"
+
+	grafanav1alpha1 "github.com/integr8ly/grafana-operator/v3/pkg/apis/integreatly/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/discovery"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("infinispan-grafana")
+
+func AddToScheme(scheme *runtime.Scheme) error {
+	return grafanav1alpha1.AddToScheme(scheme)
+}
+
+func EnsureDashboardExported(context context.Context, namespace string, mgr manager.Manager) error {
+	dc, err := discovery.NewDiscoveryClientForConfig(mgr.GetConfig())
+	if err != nil {
+		return err
+	}
+	// detect the GrafanaDashboard resource type resourceExists on the cluster
+	resourceExists, _ := k8sutil.ResourceExists(dc, grafanav1alpha1.SchemeGroupVersion.String(), grafanav1alpha1.GrafanaDashboardKind)
```

Comment:
Seems to be CRD present validation is not enough for correct working. We should ensure that Grafana operator watching expected namespace (where Grafana CR has created)

Reply:
Does this really matter? I would say just creating if the CRD is present should be sufficient. Once I figure out how to do the `SetControllerReference` it will go away if the Infinispan operator is remove as well.

Reply:
This is main Grafana CR and it presence and correct status means that Grafana correctly installed and configured inside desired namespace.